### PR TITLE
Test for issues 6561

### LINKED ISSUE #6561: Add test case for YamlProxyConfigurationSwapper

### PR DESCRIPTION
**Pull request by harvies**
Fixes #6561.

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-common/src/test/java/org/apache/shardingsphere/proxy/config/yaml/swapper/YamlProxyConfigurationSwapperTest.java [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        Assert.assertEquals(1, schemaRules.size());
+        Collection<RuleConfiguration> ruleConfigurationCollection = schemaRules.get("yamlProxyRule1");
+        Assert.assertNotNull(ruleConfigurationCollection);
+        Assert.assertEquals(1, ruleConfigurationCollection.size());
+        RuleConfiguration ruleConfiguration = ruleConfigurationCollection.iterator().next();
+        Assert.assertNotNull(ruleConfiguration);
+        Assert.assertTrue(ruleConfiguration instanceof MasterSlaveRuleConfiguration);
+    }
+
+}
```
Comment (taojintianxia) [inline shardingsphere-proxy/shardingsphere-proxy-common/src/test/java/org/apache/shardingsphere/proxy/config/yaml/swapper/YamlProxyConfigurationSwapperTest.java:256]: should add a new empty line at the end of the code for checkstyle

Comment (harvies) [reply shardingsphere-proxy/shardingsphere-proxy-common/src/test/java/org/apache/shardingsphere/proxy/config/yaml/swapper/YamlProxyConfigurationSwapperTest.java:239]: Just formatted with checkstyle.

**Comment by harvies** [comment]
> Some suggesstions:
> 
> 1. Please remove all commit and blank lines, use private method instead of it.
> 2. Please use assertThat instead of assertEquals
> 3. Please do not use Assert.xxx, use put Assert to static import

image
I resubmitted but encountered other errors. Can you help me fix it?

File: shardingsphere-proxy/shardingsphere-proxy-common/src/test/java/org/apache/shardingsphere/proxy/config/yaml/swapper/YamlProxyConfigurationSwapperTest.java [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+        assertNotNull(dataSourceParameter);
+        assertThat("url1", is(dataSourceParameter.getUrl()));
+        assertThat("username1", is(dataSourceParameter.getUsername()));
+        assertThat("password1", is(dataSourceParameter.getPassword()));
+        assertThat(1L, is(dataSourceParameter.getConnectionTimeoutMilliseconds()));
+        assertThat(2L, is(dataSourceParameter.getIdleTimeoutMilliseconds()));
+        assertThat(3L, is(dataSourceParameter.getMaxLifetimeMilliseconds()));
+        assertThat(4, is(dataSourceParameter.getMaxPoolSize()));
+        assertThat(5, is(dataSourceParameter.getMinPoolSize()));
+        assertThat(6L, is(dataSourceParameter.getMaintenanceIntervalMilliseconds()));
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-common/src/test/java/org/apache/shardingsphere/proxy/config/yaml/swapper/YamlProxyConfigurationSwapperTest.java:87]: The sequence of assertion's actual value and expected value are wrong.

File: shardingsphere-proxy/shardingsphere-proxy-common/src/test/java/org/apache/shardingsphere/proxy/config/yaml/swapper/YamlProxyConfigurationSwapperTest.java [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+
+    private void assertMetricsConfiguration(ProxyConfiguration proxyConfiguration) {
+        MetricsConfiguration metricsConfiguration = proxyConfiguration.getMetrics();
+        assertNotNull(metricsConfiguration);
+        assertThat(metricsConfiguration.getMetricsName(), is("name1"));
+        assertThat(metricsConfiguration.getHost(), is("host1"));
+        assertThat(metricsConfiguration.getPort(), is(111));
+        assertThat(metricsConfiguration.getAsync(), is(true));
+        assertThat(metricsConfiguration.getEnable(), is(true));
+        assertThat(4, is(metricsConfiguration.getThreadCount()));
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-common/src/test/java/org/apache/shardingsphere/proxy/config/yaml/swapper/YamlProxyConfigurationSwapperTest.java:118]: The actual value and expected value's sequence is wrong

**Comment by harvies** [comment]
image

**Pull request merged**